plug/pip_resolve.py

```diff
@@ -99,6 +100,48 @@ def create_package_as_root(package, dir_as_root):
 sys_platform_re = re.compile('sys_platform\s*==\s*[\'"](.+)[\'"]')
 sys_platform = sys.platform.lower()
 
+def satisfies_python_requirement(parsed_operator, parsed_python_version ):
+    # TODO: use python semver library to compare versions
+    operator_func = {
+        ">": gt,
+        "==": eq,
+        "<": lt,
+        "<=": le,
+        ">=": ge,
+        '!=': ne,
+    }[parsed_operator]
+    system_python = str(sys.version_info[0]) + '.' + str(sys.version_info[1])
+    satisfies = operator_func(float(system_python), float(parsed_python_version))
+    return satisfies
+
+def matches_python_version(requirement):
+    """Filter out requirements that should not be installed
+    in this Python version.
+    See: https://www.python.org/dev/peps/pep-0508/#environment-markers
+    """
+    python_version_re = re.compile('python_version\s*(==|<=|=>|>|<)\s*\'(.+)\'')
```

Comment:
If you do this at the module level as a constant e.g `PYTHON_VERSION_RE` then it'll only compile it once at module load time, rather than for every line of the requirements file.